pkg/endpoints/pipeline.go

```diff
@@ -166,6 +214,101 @@ func (r Resource) getPipelineRun(request *restful.Request, response *restful.Res
 	response.WriteEntity(pipelinerun)
 }
 
+func (r Resource) createPipelineRun(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter("namespace")
+
+	pipelineRunData := ManualPipelineRun{}
+	if err := request.ReadEntity(&pipelineRunData); err != nil {
+		logging.Log.Error("error parsing request body on call to createPipelineRun %s", err)
+		utils.RespondError(response, err, http.StatusBadRequest)
+		return
+	}
+
+	createResponse := r.CreatePipelineRunImpl(pipelineRunData, namespace)
+	if createResponse.CODE == 204 {
+		response.WriteHeader(http.StatusNoContent)
+	} else { // anything other than 204 is an error - RespondError
+		utils.RespondError(response, createResponse.ERROR, createResponse.CODE)
+	}
+}
+
+/*CreatePipelineRunImpl - Create a new manual pipeline run and resources in a given namespace */
```

Comment:
I think you should add some info from this comment `// Assumes you've already applied the yml: so the pipeline definition and its tasks must exist upfront.` about how we are assuming that the Pipeline & Task definitions already exist.